packages/gatsby/src/utils/parcel/compile-gatsby-files.ts

```diff
@@ -18,9 +18,7 @@ export function constructParcel(siteRoot: string): Parcel {
       `${siteRoot}/${gatsbyFileRegex}`,
       `${siteRoot}/plugins/**/${gatsbyFileRegex}`,
     ],
-    defaultConfig: require.resolve(`gatsby-parcel-config`, {
-      paths: [siteRoot],
-    }),
+    defaultConfig: require.resolve(`gatsby-parcel-config`),
```

Comment:
`gatsby-parcel-config` is dependency of `gatsby` not site, so we shouldn't try to resolve it from site context